lib/file.js

```diff
@@ -1,127 +1,123 @@
-module.exports = File
-
-var eos = require('end-of-stream')
-var EventEmitter = require('events').EventEmitter
-var FileStream = require('./file-stream')
-var inherits = require('inherits')
-var path = require('path')
-var render = require('render-media')
-var stream = require('readable-stream')
-var streamToBlob = require('stream-to-blob')
-var streamToBlobURL = require('stream-to-blob-url')
-var streamToBuffer = require('stream-with-known-length-to-buffer')
-
-inherits(File, EventEmitter)
-
-function File (torrent, file) {
-  EventEmitter.call(this)
-
-  this._torrent = torrent
-  this._destroyed = false
-
-  this.name = file.name
-  this.path = file.path
-  this.length = file.length
-  this.offset = file.offset
-
-  this.done = false
-
-  var start = file.offset
-  var end = start + file.length - 1
-
-  this._startPiece = start / this._torrent.pieceLength | 0
-  this._endPiece = end / this._torrent.pieceLength | 0
-
-  if (this.length === 0) {
-    this.done = true
-    this.emit('done')
+const eos = require('end-of-stream')
+const {EventEmitter} = require('events')
+const FileStream = require('./file-stream')
+const path = require('path')
+const render = require('render-media')
+const stream = require('readable-stream')
+const streamToBlob = require('stream-to-blob')
+const streamToBlobURL = require('stream-to-blob-url')
+const streamToBuffer = require('stream-with-known-length-to-buffer')
+
+class File extends EventEmitter {
+  constructor (torrent, file) {
+    super()
+
+    this._torrent = torrent
+    this._destroyed = false
+
+    this.name = file.name
+    this.path = file.path
+    this.length = file.length
+    this.offset = file.offset
+
+    this.done = false
+
+    const start = file.offset
+    const end = start + file.length - 1
+
+    this._startPiece = start / this._torrent.pieceLength | 0
+    this._endPiece = end / this._torrent.pieceLength | 0
+
+    if (this.length === 0) {
+      this.done = true
+      this.emit('done')
+    }
   }
-}
 
-Object.defineProperty(File.prototype, 'downloaded', {
-  get: function () {
+  get downloaded () {
     if (!this._torrent.bitfield) return 0
-    var downloaded = 0
-    for (var index = this._startPiece; index <= this._endPiece; ++index) {
+    let downloaded = 0
+    for (let index = this._startPiece; index <= this._endPiece; ++index) {
       if (this._torrent.bitfield.get(index)) {
         // verified data
-        downloaded += (index === this._endPiece) ? this._torrent.lastPieceLength : this._torrent.pieceLength
+        downloaded += index === this._endPiece ? this._torrent.lastPieceLength : this._torrent.pieceLength
```

Comment:
I think these parentheses make the code more readable.

Reply:
I'd rather keep them